python-avd/pyavd/_eos_designs/structured_config/underlay/router_pim_sparse_mode.py

```diff
@@ -19,53 +20,38 @@ class RouterPimSparseModeMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def router_pim_sparse_mode(self: AvdStructuredConfigUnderlayProtocol) -> dict | None:
+    @structured_config_contributor
+    def router_pim_sparse_mode(self: AvdStructuredConfigUnderlayProtocol) -> None:
         """
-        Return structured config for router_pim_sparse_mode.
+        Set the structured config for router_pim_sparse_mode.
 
         Used for to configure multicast RPs for the underlay
         """
         if not self.shared_utils.underlay_multicast or not self.inputs.underlay_multicast_rps:
-            return None
+            return
 
-        rp_addresses = []
-        anycast_rps = []
         for rp_entry in self.inputs.underlay_multicast_rps:
-            rp_address = {"address": rp_entry.rp}
+            rp_address = EosCliConfigGen.RouterPimSparseMode.Ipv4.RpAddressesItem(address=rp_entry.rp)
             if rp_entry.groups:
                 if rp_entry.access_list_name:
-                    rp_address["access_lists"] = [rp_entry.access_list_name]
+                    rp_address.access_lists.append(rp_entry.access_list_name)
                 else:
-                    rp_address["groups"] = rp_entry.groups._as_list()
+                    for group in rp_entry.groups:
+                        rp_address.groups.append(group)
 
-            rp_addresses.append(rp_address)
+            self.structured_config.router_pim_sparse_mode.ipv4.rp_addresses.append(rp_address)
 
             if len(rp_entry.nodes) < 2 or self.shared_utils.hostname not in rp_entry.nodes or self.inputs.underlay_multicast_anycast_rp.mode != "pim":
                 continue
 
             # Anycast-RP using PIM (default)
-            anycast_rps.append(
-                {
-                    "address": rp_entry.rp,
-                    "other_anycast_rp_addresses": [
-                        {
-                            "address": get(self.shared_utils.get_peer_facts(node.name), "router_id", required=True),
-                        }
-                        for node in rp_entry.nodes
-                    ],
-                },
+            other_anycast_rp_addresses = EosCliConfigGen.RouterPimSparseMode.Ipv4.AnycastRpsItem.OtherAnycastRpAddresses()
+            for node in rp_entry.nodes:
+                other_anycast_rp_addresses.append(
+                    EosCliConfigGen.RouterPimSparseMode.Ipv4.AnycastRpsItem.OtherAnycastRpAddressesItem(
+                        address=get(self.shared_utils.get_peer_facts(node.name), "router_id", required=True)
+                    )
+                )
```

Comment:
Given that you just create the Item and never touch it again we can use append_new here
```suggestion
                other_anycast_rp_addresses.append_new(address=get(self.shared_utils.get_peer_facts(node.name), "router_id", required=True))
```